api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java

```diff
@@ -1779,9 +1779,9 @@ public Concept getConceptByName(final String name) {
 		criteria.add(Restrictions.or(Restrictions.eq("locale", locale), Restrictions.like("locale", language)));
 		
 		if (Context.getAdministrationService().isDatabaseStringComparisonCaseSensitive()) {
```

Comment:
@dkayiwa , Yes the behavior is changed. But if this condition evaluates to true we are supposed to search case sensitively ? 
We don't get this issue on MySQL because its collation is set to utf8_general_ci that is it is case insensitive. So, if you search for one of the concept with name "Systolic blood pressure" in db with query : 
**SELECT * FROM concept_name where name='SYSTOLIC BLOOD PRESSURE';**
This will give result in case of MySQL but not in case of PostgreSQL. This is the reason why no demo data is being added in case of PostgreSQL.

Reply:
Can you search the codebase for other cases of `isDatabaseStringComparisonCaseSensitive` and see what they do when the value is true?

Reply:
I found it [here](https://github.com/openmrs/openmrs-core/blob/77e5e8c4315a98f079ecab9ad9429b62cfd461a7/api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java#L351) , [here](https://github.com/openmrs/openmrs-core/blob/77e5e8c4315a98f079ecab9ad9429b62cfd461a7/api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java#L1035) , [here](https://github.com/openmrs/openmrs-core/blob/77e5e8c4315a98f079ecab9ad9429b62cfd461a7/api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java#L1043) and [here](https://github.com/openmrs/openmrs-core/blob/1c21cb563496bbfeb58133b0ad2ea8b39121783e/api/src/main/java/org/openmrs/api/db/hibernate/PatientSearchCriteria.java#L282) . And apparently it appears to me it is supposed to behave opposite to what it is named. If it returns true then we are doing a case insensitive search. So, my implementation is not right.

Reply:
There are two solutions coming in my mind : either I set the global property GP_CASE_SENSITIVE_DATABASE_STRING_COMPARISON this to true before Ref. Demo Data setup or I change all the values in Ref. Demo Data to match with DB ?

Reply:
That is exactly what i wanted you to see! 😊 

Which of the two do you think would be better?

Reply:
Apparently, I think setting global property would be a better way. This way there would be minimal changes. Also, default value of this property is supposed to be true. There must be somewhere that it is being switched to false while calling the ReferenceDemoData Activator.

Reply:
After logging in, does it have a value of false?

Reply:
Yes, you can check in manage global properties section. What do you think should I go ahead with setting global property approach ?

Reply:
Isn't setting the global property simply hiding a bug?

Reply:
If we hold a use case where concept with name say "DENGUE" and "dengue" need to be treated separately, then it will turn out to be a data inconsistency bug and may lead to wrong data insertions. Then correcting the data in Ref. Demo Data module will be the ideal approach.

Reply:
I agree. So can you go with the ideal approach?

Reply:
I am on it :slightly_smiling_face: